wifiutils/src/main/java/com/thanosfisherman/wifiutils/ConnectorUtils.java

```diff
@@ -776,9 +776,9 @@ static void reenableAllHotspots(@Nullable WifiManager wifi) {
     }
 
     @Nullable
-    static ScanResult matchScanResultSsid(@NonNull String ssid, @NonNull Iterable<ScanResult> results) {
+    static ScanResult matchScanResultSsid(@NonNull String ssid, @NonNull Iterable<ScanResult> results, boolean mPatternMatch) {
         for (ScanResult result : results) {
-            if (Objects.equals(result.SSID, ssid)) {
+            if (mPatternMatch == true ? result.SSID.indexOf(ssid)> -1 : Objects.equals(result.SSID, ssid)) {
```

Comment:
Maybe here you know better solution to check if network is found

Reply:
This line can use some refactoring. Try this instead
`if (mPatternMatch ? result.SSID.startsWith(ssid) : Objects.equals(result.SSID, ssid))`